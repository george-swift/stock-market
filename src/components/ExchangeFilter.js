import PropTypes from 'prop-types';
import { EXCHANGE } from '../constants';

export default function ExchangeFilter({ filterExchange }) {
  const exchange = ['All Markets', ...EXCHANGE];

  const handleFilterChange = (e) => {
    const { value } = e.target;
    filterExchange(value);
  };

  return (
    <select
      className="form-select my-3"
      onChange={handleFilterChange}
    >
      {
        exchange.map((category) => (
          <option key={category} value={category}>
            {category}
          </option>
        ))
      }
    </select>
  );
}

ExchangeFilter.propTypes = {
  filterExchange: PropTypes.func.isRequired,
};
