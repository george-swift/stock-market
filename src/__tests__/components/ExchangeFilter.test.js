import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import ExchangeFilter from '../../components/ExchangeFilter';

const changeMarket = jest.fn(() => {});

describe('Snapshot of the ExchangeFilter Component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<ExchangeFilter filterExchange={changeMarket} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Querying the ExchangeFilter component', () => {
  it('should display exchange markets in three main categories', () => {
    render(<ExchangeFilter filterExchange={changeMarket} />);

    const categories = screen.getAllByRole('option');
    expect(categories).toHaveLength(3);
    expect(categories.some((option) => option.textContent.match('All Markets'))).toBeTruthy();
    expect(categories.some((option) => option.textContent.match('Nasdaq Global Select'))).toBeTruthy();
    expect(categories.some((option) => option.textContent.match('New York Stock Exchange'))).toBeTruthy();
  });
});
