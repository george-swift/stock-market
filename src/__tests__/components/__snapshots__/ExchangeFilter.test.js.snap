// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of the ExchangeFilter Component should render correctly 1`] = `
<select
  className="form-select my-3"
  onChange={[Function]}
>
  <option
    value="All Markets"
  >
    All Markets
  </option>
  <option
    value="New York Stock Exchange"
  >
    New York Stock Exchange
  </option>
  <option
    value="Nasdaq Global Select"
  >
    Nasdaq Global Select
  </option>
</select>
`;
